File: tarea7-intro-jgonzalez/tarea7-intro-jgonzalez/mult-matrices-rect.py
def matrices(m1,m2):
	if isinstance(m1,list) and isinstance(m2,list):
		multiplicar(m1,m2,0,0,0,0,[])
	else:
		'No es una matriz'
def multiplicar(m1,m2,i,i2,i3,suma,new):
	if len(m1) == i:
		print(new)
	elif len(m2) == i2 and len(m2[0]) == i3+1:
		return multiplicar(m1,m2,i+1,0,0,0,new+[[suma]])
	elif len(m2) == i2:
		return multiplicar(m1,m2,i,0,i3+1,0,new+[[suma]])
	else:
		return multiplicar(m1,m2,i,i2+1,i3,suma + m1[i][i2]*m2[i2][i3],new)

File: tarea7-intro-jgonzalez/tarea7-intro-jgonzalez/mult-matrices-rect.java
package com.company;

import java.util.Scanner;

public class Main {

        int[][] A = { { 1,2,3,4 }, { 5,6,7,8 } };
        int[][] B = { { 1,2 }, { 3,4}, {5,6}, {7,8} };

        public static int[][] multiplicar(int[][] A, int[][] B) {

            int aFilas = A.length;
            int aColumnas = A[0].length;
            int bFilas = B.length;
            int bColumnas = B[0].length;

            if (aColumnas != bFilas) {
                throw new Error("las matrices no son rectangulares");
            }

            int[][] C = new int[aFilas][bColumnas];
            for (int i = 0; i < aFilas; i++) {
                for (int j = 0; j < bColumnas; j++) {
                    C[i][j] = 0;
                }
            }

            for (int i = 0; i < aFilas; i++) {
                for (int j = 0; j < bColumnas; j++) {
                    for (int k = 0; k < aColumnas; k++) {
                        C[i][j] += A[i][k] * B[k][j];
                    }
                }
            }

            return C;
        }

        public static void main(String[] args) {


            Main matrix = new Main();

            int[][] result = multiplicar(matrix.A, matrix.B);

            for (int i = 0; i < 2; i++) {
                for (int j = 0; j < 2; j++)
                    System.out.print(result[i][j] + " ");
                System.out.println();
            }
        }
    }